Extensions/DialogueTree/dialoguetools.js

```diff
@@ -0,0 +1,538 @@
+    return this.commandParameters.length - 1;
+  }
+  return 0;
+};
+
+/**
+ * Expression to get a command parameter in any command with parameters that has been caugth by a commandIsCalled condition
+ * @param {number} paramIndex The index of the parameter you want to get. If left empty, the expression will get the first parameter by default.
+ */
+gdjs.dialogueTree.getCommandParameter = function(paramIndex) {
+  if (
+    this.commandParameters &&
+    this.commandParameters.length >= paramIndex + 1
+  ) {
+    var returnedParam = this.commandParameters[paramIndex + 1];
+    return returnedParam ? returnedParam : "";
+  }
+  return "";
+};
+
+/**
+ * Condition to catch <<commands>> and <<commands with parameters>> from the current Dialogue Line.
+ * Use this to trigger events in the event sheet that relate to the story you are telling throught the dialogue.
+ * @param {string} command The command you want to check for being called. Write it without the <<>>
+ */
+gdjs.dialogueTree.commandIsCalled = function(command) {
+  var { commandCalls, clipTextEnd, dialogueText } = gdjs.dialogueTree;
+
+  if (this.pauseScrolling || !commandCalls) return false;
+  return this.commandCalls.some(function(call, index) {
+    if (clipTextEnd < call.time) return false;
+    if (call.cmd === "wait" && clipTextEnd !== dialogueText.length) {
+      gdjs.dialogueTree.pauseScrolling = true;
+      setTimeout(function() {
+        gdjs.dialogueTree.pauseScrolling = false;
+        commandCalls.splice(index, 1);
+      }, parseInt(call.params[1]));
+    }
+    if (call.cmd === command) {
+      gdjs.dialogueTree.commandParameters = call.params;
+      commandCalls.splice(index, 1);
+      return true;
+    }
+  });
+};
+
+// Internal method to allow for capping option selection
+gdjs.dialogueTree._normalizedOptionIndex = function(optionIndex) {
+  if (optionIndex >= this.options.length) optionIndex = this.options.length - 1;
+  if (optionIndex < 0) optionIndex = 0;
+  return optionIndex;
+};
+
+// Internal method to allow for cycling option selection
+gdjs.dialogueTree._cycledOptionIndex = function(optionIndex) {
+  if (optionIndex >= this.options.length) optionIndex = 0;
+  if (optionIndex < 0) optionIndex = this.options.length - 1;
+  return optionIndex;
+};
+
+/**
+ * Expression to get the text of an option the player can select.
+ * Use this with lineOptionsCount to render options for the player when a line of the Options type is parsed
+ * @param {number} optionIndex The index of the option you want to get
+ */
+gdjs.dialogueTree.getLineOption = function(optionIndex) {
+  if (!this.options.length) return [];
+  optionIndex = gdjs.dialogueTree._normalizedOptionIndex(optionIndex);
+  return this.options[optionIndex];
+};
+
+// Expression to get the number of options that are presented to the player during the parsing of an Options type line
+gdjs.dialogueTree.lineOptionsCount = function() {
+  if (this.options.length) {
+    return this.optionsCount;
+  }
+};
```

Comment:
You will return undefined if we reach this point, this is dangerous as this will probably break things down the line and create hard to catch, hard to debug bugs. Always always always return the same type, surely 0 here.